packages/hierarchies-react/src/test/UseSelectionHandler.test.tsx

```diff
@@ -0,0 +1,261 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import { expect } from "chai";
+import sinon from "sinon";
+import { renderHook } from "@testing-library/react";
+import { PresentationHierarchyNode, PresentationInfoNode, PresentationTreeNode } from "../presentation-hierarchies-react";
+import { SelectionChangeType, SelectionMode, useSelectionHandler } from "../presentation-hierarchies-react/UseSelectionHandler";
+
+describe("useSelectionHandler", () => {
+  const selectNodesStub = sinon.stub<[Array<string>, SelectionChangeType], void>();
+
+  const createMouseEvent = (key?: "Shift" | "Ctrl") => {
+    return { shiftKey: key === "Shift", ctrlKey: key === "Ctrl" } as unknown as React.MouseEvent<HTMLDivElement, MouseEvent>;
+  };
+
+  const createKeyboardEvent = (mainKey?: " " | "Spacebar" | "Enter", extraKey?: "Shift" | "Ctrl") => {
+    return { key: mainKey, shiftKey: extraKey === "Shift", ctrlKey: extraKey === "Ctrl" } as unknown as React.KeyboardEvent<HTMLDivElement>;
+  };
```

Comment:
Maybe instead of faking events we could render hook inside test component and perform actions using `user-event` library? It has more logic to replicated how interactions work in browser. Something like this: 

```ts
function TestComponent({ rootNodes, onSelect, selectionMode }) {
  const { onNodeKeyDown, onNodeClick } = useSelectionHandler({ rootNodes, onSelect, selectionMode });
  return <>
    {rootNodes.map((node) => {
      return <div 
        onClick={(e) => onNodeClick(node.id, !node.isSelected, e)} 
        onKeyDown={(e) => onNodeKeyDown(node.id, !node.isSelected, e)}
      >
        {node.label}
      </div>;
    })}
  </>
}

```

Reply:
Switched to `user-event` to simulate clicks.